pkg/util/dynamichelper/dynamichelper.go

```diff
@@ -289,3 +422,132 @@ func makeURLSegments(gvr *schema.GroupVersionResource, namespace, name string) (
 
 	return url
 }
+
+func notFound(err error) bool {
+	if err == nil || kerrors.IsNotFound(err) || strings.Contains(strings.ToLower(err.Error()), "notfound") {
+		return true
+	}
+	return false
+}
+
+// mergeGK takes the existing (old) and desired (new) objects. It checks the
+// the interested fields in the *new* object to see if an update is necessary,
+// fixes up the *old* object if needed, and returns the difference for
+// debugging purposes. The reason for using *old* as basis is that the *old*
+// object are changed by gatekeeper binaries and the changes must be kept.
+func mergeGK(old, new kruntime.Object) (kruntime.Object, bool, string, error) {
+	if reflect.TypeOf(old) != reflect.TypeOf(new) {
+		return nil, false, "", fmt.Errorf("types differ: %T %T", old, new)
+	}
+	expected := old.DeepCopyObject()
+
+	// 1. Set defaults on old.  This gets rid of many false positive diffs.
+	scheme.Scheme.Default(expected)
+
+	// 2. Do fix-ups on a per-Kind basis.
+	hasChanged := false
+	switch new.(type) {
+	case *appsv1.Deployment:
+		hasChanged = handleDeployment(new, expected)
+	case *admissionregistrationv1.ValidatingWebhookConfiguration:
+		hasChanged = handleValidatingWebhook(new, expected)
+	case *admissionregistrationv1.MutatingWebhookConfiguration:
+		hasChanged = handleMutatingWebhook(new, expected)
+	}
+	var diff string
+	if _, ok := expected.(*corev1.Secret); !ok { // Don't show a diff if kind is Secret
+		diff = cmp.Diff(old, expected)
+	}
+	return expected, hasChanged, diff, nil
+}
+
+func handleDeployment(new, expected kruntime.Object) bool {
```

Comment:
This function should be named hasDeploymentChanged as it returns a bool

Reply:
it does not just check if it changed, it also copy stuff around, the **"has"** will be confusing then